test/e2e/tests/test_sagemaker_endpoint_autoscaling.py

```diff
@@ -0,0 +1,201 @@
+
+    if k8s.get_resource_exists(policy_reference):
+        _, deleted = k8s.delete_custom_resource(policy_reference)
+        assert deleted
+
+
+@service_marker
+@pytest.mark.canary
+class TestSageMakerEndpointAutoscaling:
+    def get_sagemaker_scalable_target_description(
+        self, applicationautoscaling_client, resource_id: str, expectedTargets: int
+    ):
+        try:
+            targets = applicationautoscaling_client.describe_scalable_targets(
+                ServiceNamespace="sagemaker",
+                ResourceIds=[resource_id],
+            )
+
+            assert len(targets["ScalableTargets"]) == expectedTargets
+            return targets["ScalableTargets"]
+        except BaseException:
+            logging.error(
+                f"ApplicationAutoscaling could not find a scalableTarget for the resource {resource_id}"
+            )
+            return None
+
+    def get_sagemaker_scaling_policy_description(
+        self, applicationautoscaling_client, resource_id: str, expectedPolicies: int
+    ):
+        try:
+            policies = applicationautoscaling_client.describe_scaling_policies(
+                ServiceNamespace="sagemaker",
+                ResourceId=resource_id,
+            )
+            assert len(policies["ScalingPolicies"]) == expectedPolicies
+            return policies["ScalingPolicies"]
+        except BaseException:
+            logging.error(
+                f"ApplicationAutoscaling could not find a scalingPolicy for the resource {resource_id}"
+            )
+            return None
+
+    def test_smoke(self, applicationautoscaling_client, generate_sagemaker_policy):
+        (
+            resource_id,
+            target_reference,
+            policy_resource,
+            policy_reference,
+        ) = generate_sagemaker_policy
+
+        target_description = self.get_sagemaker_scalable_target_description(
+            applicationautoscaling_client, resource_id, 1
+        )
+        assert target_description is not None
+
+        policy_description = self.get_sagemaker_scaling_policy_description(
+            applicationautoscaling_client, resource_id, 1
+        )
+        assert policy_description is not None
+
+        assert k8s.get_resource_arn(policy_resource) is not None
+        assert policy_description[0] is not None
+        assert (
+            k8s.get_resource_arn(policy_resource) == policy_description[0]["PolicyARN"]
+        )
+
+        # Delete the Resource
+
+        _, deleted = k8s.delete_custom_resource(policy_reference)
+        assert deleted is True
+
+        _, deleted = k8s.delete_custom_resource(target_reference)
+        assert deleted is True
+
+        target_description = self.get_sagemaker_scalable_target_description(
+            applicationautoscaling_client, resource_id, 0
+        )
+
+        # TODO: Ideally this check should pass after line 190 itself; but it requires the scalabletarget to be deleted too.
```

Comment:
remove TODO and assert these descriptions to `None`? to see delete succeded

Reply:
the descriptions will not be none, it is an empty list. The size assertion is happening in the method. Can you confirm this is enough.

Reply:
I want to keep the TODO because this is something that should be confirmed with the application-autoscaling team.